File: sach/de3335.py
//DE3335.PY - Tran Huu Nam - hsgtin.vn - 2024
import bisect
n, m = 0, 0
a = []
def main():
    global n, m, a
    n = int(input())
    a = [int(input()) for _ in range(n)]
    a.sort()
    m = int(input())
    for _ in range(m):
        s, f = map(int, input().split())
        x = bisect.bisect_left(a, s)
        y = bisect.bisect_right(a, f) - 1
        if 0 <= x < n and 0 <= y < n:
            print(y - x + 1)
        else:
            print(0)
if __name__ == "__main__":
    main()

File: sach/de3335.cpp
//DE3335.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
int n,m,s,f,x,y; int a[100005];
int main() {
    cin>>n;
    for (int i=1; i<=n; i++) cin>>a[i];
    sort(a+1,a+n+1);
    cin>>m;
    for (int i=1; i<=m; i++) {
        cin>>s>>f;
        x=lower_bound(a+1,a+n+1,s)-a;
        y=upper_bound(a+1,a+n+1,f)-a-1;
        if (x>=1 && x<=n && y>=1 && y<=n)
            cout<<y-x+1<<endl;
        else
            cout<<0<<endl;
    }
    return 0;
}

